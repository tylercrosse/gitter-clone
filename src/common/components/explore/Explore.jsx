import React from 'react';
import PropTypes from 'prop-types';
import { Link }    from 'react-router';
import { connect } from 'react-redux';
// import                  './explore.scss';

export const Explore = (props) => (
  <div className="explore">
    <header className="explore-header">
      <div className="explore-header-logo">Gitter-clone</div>
      <nav className="explore-header-nav" />
      <h1>Explore</h1>
    </header>
    <section className="explore-page-description">
      <h3>Take a peak at some of the conversations</h3>
    </section>
    <section className="convo-card-list-section">
      {ConvoCards(props)}
    </section>
  </div>
);

export const ConvoCards = (props) => {
  const renderConvoCard = (convo) => (
    <Link
    key={convo.id}
    className="convo-card"
    to={'/' + convo.name}
    >
      <header className="convo-card-header">
        <img className="convo-card-avatar" src={'http://i.pravatar.cc/30?u=' + convo.name} alt={convo.name} />
        <h3>{convo.name}</h3>
      </header>
      <footer className="convo-card-footer">
        <div className="convo-card-info">X{/* convo.messages.length */} msgs</div>
        <div className="convo-card-action">
          <button>view</button>
        </div>
      </footer>
    </Link>
  );

  return (
    <div className="convo-card-list">
      {Object.values(props.convos).map(renderConvoCard)}
    </div>
  );
};

Explore.propTypes = {
  convos: PropTypes.object.isRequired,
};

export const mapStateToProps = (state) => ({
  convos: state.convos
});

export default connect(mapStateToProps)(Explore);
