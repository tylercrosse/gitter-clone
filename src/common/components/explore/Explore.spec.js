import React       from 'react';
import renderer    from 'react-test-renderer';
import { Explore } from './Explore';

const setup = (propOverrides) => {
  const props = Object.assign({
    convos: {
      '5898b14f6047fd8db8b4e3ba': {
        id: '5898b14f6047fd8db8b4e3ba',
        updatedAt: '2017-02-06T17:24:31.593Z',
        createdAt: '2017-02-06T17:24:31.593Z',
        name: 'chat',
        messages: ['1'],
        __v: 0
      },
      '5898b15d6047fd8db8b4e3bb': {
        id: '5898b15d6047fd8db8b4e3bb',
        updatedAt: '2017-02-06T17:24:45.657Z',
        createdAt: '2017-02-06T17:24:45.657Z',
        name: 'chat2',
        messages: ['1', '2', '3'],
        __v: 0
      }
    }
  }, propOverrides);

  return {
    props
  };
};

describe('<ConvoItem />', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const component = <Explore {...props} />;
    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
