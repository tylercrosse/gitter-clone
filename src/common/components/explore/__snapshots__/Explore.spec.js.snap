// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConvoItem /> should render correctly 1`] = `
<div
  className="explore"
>
  <header
    className="explore-header"
  >
    <div
      className="explore-header-logo"
    >
      Gitter-clone
    </div>
    <nav
      className="explore-header-nav"
    />
    <h1>
      Explore
    </h1>
  </header>
  <section
    className="explore-page-description"
  >
    <h3>
      Take a peak at some of the conversations
    </h3>
  </section>
  <section
    className="convo-card-list-section"
  >
    <div
      className="convo-card-list"
    >
      <a
        className="convo-card"
        onClick={[Function]}
        style={Object {}}
      >
        <header
          className="convo-card-header"
        >
          <img
            alt="chat"
            className="convo-card-avatar"
            src="http://i.pravatar.cc/30?u=chat"
          />
          <h3>
            chat
          </h3>
        </header>
        <footer
          className="convo-card-footer"
        >
          <div
            className="convo-card-info"
          >
            X
             msgs
          </div>
          <div
            className="convo-card-action"
          >
            <button>
              view
            </button>
          </div>
        </footer>
      </a>
      <a
        className="convo-card"
        onClick={[Function]}
        style={Object {}}
      >
        <header
          className="convo-card-header"
        >
          <img
            alt="chat2"
            className="convo-card-avatar"
            src="http://i.pravatar.cc/30?u=chat2"
          />
          <h3>
            chat2
          </h3>
        </header>
        <footer
          className="convo-card-footer"
        >
          <div
            className="convo-card-info"
          >
            X
             msgs
          </div>
          <div
            className="convo-card-action"
          >
            <button>
              view
            </button>
          </div>
        </footer>
      </a>
    </div>
  </section>
</div>
`;
